modules/core/Scene.cpp

```diff
@@ -152,5 +177,46 @@ namespace rt {
 	{ 
 		return intersect(ray) ? ray.t : std::numeric_limits<double>::infinity();
 	}
-	
+
+    bool CScene::save(const std::string &fileName, const Mat& image) const {
+        std::ofstream ofs(fileName, std::ios::binary);
+        if(!ofs.is_open()){
+            return false;
+        }
+        if(image.empty()){
```

Comment:
I still think whether it is really makes sense to save the rendered frame in binary format... How much time can we win? Does this time somehow comparable to the rendering time? 
I think about a situation, when render takes half an hour, and user suddenly closes the image and forgets to save it. Then all 30 minutes are wasted. But if a user knows that the rendered image is cached - then user could just take the cached image as it is and copy it into a user folder (in this case the cached images may work as a backup)

Reply:
Actually, it only makes sense to store in binary format if we're doing the whole caching inside of the `scene.render(args)` method because it does increase the overall render time. Otherwise, if we decide that the user calls this method elsewhere, then he's aware the streaming operations can be slow on high-res images + no stitching of images so much easier.
 Here are my two remarks:
1) If the user accidentally closes the image, they can always restore it using the `getLastRenderedFrame()` method right? The downside is that they would have to do it in a programatic way (getLastRenderedFrame() -> writeImage()), rather than a copy and paste.
2) I really like the backup idea. It makes a lot of sense esp. with very large scenes.